### DISCUSSION TITLE
PickFirst Selection Set

### DISCUSSION THREAD

**Discussion by kwollaston** [CEXT-Dan/PyRx#83] [Q&A] [answered]
I am trying to adjust a command to accept a selection prior to execution of the command. I see there is something called PickFirstSelectionSet in the docs but I've been unable to access it. Can you share an example if you don't mind? I scoured the examples for this but couldn't find one and wasn't able to piece it together from the docs.

**Comment by CEXT-Dan** [comment]
If I am understanding correctly, one of these samples may help

You may register a command with USEPICKSET or’d with REDRAW
And use "I"  The PICKFIRST set  

You can get the previous selection set.

Additionally, there’s a handy function
ids = Ed.Editor.getCurrentSelectionSet()
ids = Ed.Core.getCurrentSelectionSet()

is a wrapper around acedGetCurrentSelectionSet
https://help.autodesk.com/view/OARX/2023/ENU/?guid=OARX-RefGuide-acedGetCurrentSelectionSet_AcDbObjectIdArray_

```Python
from pyrx_impx import Rx, Ge, Gi, Db, Ap, Ed, Ax
import traceback


def PyRxCmd_DOITPKF(CmDFlags = Ap.CmdFlags.USEPICKSET|Ap.CmdFlags.REDRAW):
    try:
        ss = Ed.Editor.ssget("I", None, None)
        print("Woohoo ", ss)
        print("Length =  ", len(ss[1].objectIds()))
    except Exception:
        traceback.print_exc()

def PyRxCmd_DOITSP():
    try:
        ss = Ed.Editor.selectPrevious()
        print("Woohoo ", ss)
        print("Length =  ", len(ss[1].objectIds()))
        
    except Exception:
        traceback.print_exc()
        
def PyRxCmd_DOITCSS():
    try:
        
        ids = Ed.Editor.getCurrentSelectionSet()
        print("Woohoo ", len(ids))
        
    except Exception:
        traceback.print_exc()

```

**Reply by CEXT-Dan** [reply, marked as answer]
.NET has a wrapper for acedssgetfirst, I have added this as 
Ed.Editor.selectImplied()
It will be available next release (v1.3.026)

```Python
def PyRxCmd_DOITIMP(CmDFlags=Ap.CmdFlags.USEPICKSET | Ap.CmdFlags.REDRAW):
    try:
        ss = Ed.Editor.selectImplied()
        print("Woohoo ", ss)
        print("Length =  ", len(ss[1].objectIds()))
    except Exception:
        traceback.print_exc()
```

**Reply by kwollaston** [reply]
Thank you, I tried selectImplied as well actually!

**Comment by Hairo** [comment]
Worth to mention that just setting `CmDFlags=Ap.CmdFlags.USEPICKSET` should [work with your existing code](https://forums.autodesk.com/t5/net/filter-pickfirst-selectionset/m-p/5705796#M45141), no need to use `selectImplied` (unless you explicitly want it), so something like below also works and can use filters too:

```python
def PyRxCmd_DOIT(CmDFlags=Ap.CmdFlags.USEPICKSET | Ap.CmdFlags.REDRAW):
    try:
        sfilter = [(Db.DxfCode.kDxfStart, "LWPOLYLINE")]
        ss = Ed.Editor.selectPrompt("\nAdd polyline: ", "\nRemove polyline: ", sfilter)
        print(ss)
        print(len(ss[1].objectIds())
    except Exception:
        traceback.print_exc()
```
